utils/auth/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crypto::SignatureType;
+use jsonrpc_v2::Error as JsonRpcError;
+use jsonwebtoken::errors::Result as JWTResult;
+use jsonwebtoken::{decode, encode, DecodingKey, EncodingKey, Header, Validation};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use wallet::KeyInfo;
+
+/// constant string that is used to identify the JWT secret key in KeyStore
+pub const JWT_IDENTIFIER: &str = "auth-jwt-private";
+/// Admin permissions
+pub const ADMIN: [&str; 4] = ["read", "write", "sign", "admin"];
+/// Signing permissions
+pub const SIGN: [&str; 3] = ["read", "write", "sign"];
+/// Writing permissions
+pub const WRITE: [&str; 2] = ["read", "write"];
+/// Reading permissions
+pub const READ: [&str; 1] = ["read"];
```

Comment:
This could be a lot better (enums with a value of priority or something) but can always change later I guess

Reply:
Yeah I think this was from Jaden's existing implementation. I think can be just changed later for now